server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/ProvisioningTest.java

```diff
@@ -97,7 +98,20 @@ public void provisionMailboxesShouldNotDuplicateMailboxByName() throws Exception
             .body(NAME, equalTo("mailboxes"))
             .body(ARGUMENTS + ".list", hasSize(5))
             .body(ARGUMENTS + ".list.name", hasItems(DefaultMailboxes.DEFAULT_MAILBOXES.toArray()));
+    }
+
+    @Test
+    public void provisionMailboxesShouldRegisterThem() throws Exception {
+        String token = HttpJmapAuthentication.authenticateJamesUser(baseUri(), USER, PASSWORD).serialize();
+
+        with()
+            .header("Authorization", token)
+            .body("[[\"getMailboxes\", {}, \"#0\"]]")
+            .post("/jmap");
 
+        assertThat(jmapServer.getProbe(MailboxProbeImpl.class)
+            .listSubscriptions(USER))
+            .containsOnlyElementsOf(DefaultMailboxes.DEFAULT_MAILBOXES);
```

Comment:
why `Only` ?

Reply:
Order of elements? We care about the elements in there, not their order, hence it is a containOnly.

Reply:
You got me wrong ; we don't need order nor strict equality so `contains` should be enough. My question was : why would we need to check there's nothing else returned ?

Reply:
Hint: that is what I first wrote ;-)

Reply:
it's a general principle : don't assert more than the test description assumes